### DISCUSSION TITLE
empty field values being sent in http post

### DISCUSSION THREAD

**Discussion by khyberkanwar** [Budibase/budibase#7335] [Help]
Hi,

I am new to budibase and working on my 1st sample app.

I am having difficulty referencing form field values while making an outbound REST api call. Have tried several different ways to reference field values (see capture 1) but nothing has worked.  Please advise on how to reference fields values.

My sample app is supposed to construct json using the form data and perform a http post when the save button is clicked. The http post executes successfully but empty field values are sent out.

Here is the output that is being received:
{
  "parm1": "",
  "parm2": "",
  "parm3": "",
  "parm4": ""
}

Here is the screen capture (1) from data view of my outbound rest call:
<img width="1067" alt="image" src="https://user-images.githubusercontent.com/7241760/185203533-b9eb46cc-9ae6-4d1a-a15c-17b9a8bc577e.png">

Here is how app looks (2):
<img width="942" alt="image" src="https://user-images.githubusercontent.com/7241760/185203410-aa67f164-b1c6-40be-9b67-0c2a37e3280a.png">

Here is the screen capture (3) from design view of the screen:
<img width="1442" alt="image" src="https://user-images.githubusercontent.com/7241760/185203702-e39701df-6827-48cb-9fa4-eed796a3af0d.png">

You help will be much appreciated!

**Comment by melohagan** [comment]
Hi @khyberkanwar 

I think you've mis-understood how the bindings work. You have to pass through the form field binding values from the design section: https://docs.budibase.com/docs/rest-bindings#data-provider-applications-provider
